community/kernel/src/main/java/org/neo4j/kernel/impl/util/HexPrinter.java

```diff
@@ -307,7 +307,7 @@ public static String hex( ByteBuffer bytes, int offset, int length )
      */
     public static String hex( ByteBuffer bytes )
     {
-        return hex( bytes, 0, bytes.position() );
+        return hex( bytes, 0, bytes.limit() );
```

Comment:
This impl is different from `HexPrinter`'s `append` impl. It might cause some confusion to other users. Why we have to call `bytes.flip()` before we call this method? This method will not do any writing to the given buffer.
```
     /**
      * Append the bytes in the byte buffer, starting from position 0, ending at the buffer's current position,
      * into print stream
      * @param bytes
      * @return
      */
     public HexPrinter append( ByteBuffer bytes )
```